### PR TITLE
Proto serializer

### PR DESCRIPTION
**Pull request by mlw**
* Add enhanced protobuf schema
* Add protobuf serializer
* Reorganize common serializer code into a serializer utilities file
* Remove "rules_proto_grpc" from the workspace, replace with "com_google_protobuf"

### CODE REVIEW

File: Source/santad/EventProviders/EndpointSecurity/EnrichedTypes.h [hunk 1]
```diff
@@ -32,13 +32,30 @@ namespace santa::santad::event_providers::endpoint_security {
```
Comment (mlw) [inline Source/santad/EventProviders/EndpointSecurity/EnrichedTypes.h:32]: Note: Almost all changes in this file are adding move constructors, deleting copy constructors, and adding accessors for enriched members

**Review by pmarkowsky — CHANGES_REQUESTED** [review]

File: Source/common/TestUtils.mm [hunk 2]
```diff
@@ -62,6 +77,10 @@ es_process_t MakeESProcess(es_file_t *file, audit_token_t tok, audit_token_t par
     .audit_token = tok,
     .ppid = audit_token_to_pid(parent_tok),
     .original_ppid = audit_token_to_pid(parent_tok),
+    .group_id = 111,
+    .session_id = 222,
+    .is_platform_binary = true,
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/common/TestUtils.mm:82]: No. Currently only new tests I've added use function.

File: WORKSPACE [hunk 3]
```diff
@@ -14,27 +14,17 @@ http_archive(
...
-
-rules_proto_dependencies()
-
-rules_proto_toolchains()
-
-load("@rules_proto_grpc//objc:repositories.bzl", rules_proto_grpc_objc_repos = "objc_repos")
+load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
 
-rules_proto_grpc_objc_repos()
+protobuf_deps()
```
[the remark this answers is not in the document]
Comment (mlw) [reply WORKSPACE:27]: In the PR adopting fsspool I've left a comment.

File: Source/common/TestUtils.mm (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmarkowsky) [reply Source/common/TestUtils.mm:82]: Yeah I realized afterwards that this was only for testing that we do handle serializing the value.

File: Source/common/santa.proto [hunk 4]
```diff
@@ -11,135 +11,271 @@ import "google/protobuf/timestamp.proto";
 
 option objc_class_prefix = "SNTPB";
 
-package santa;
+package santa.pb;
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/common/santa.proto]: Done

File: Source/santad/Logs/EndpointSecurity/Serializers/Protobuf.mm [hunk 5]
```diff
@@ -0,0 +1,609 @@
...
+
+  if (msg.es_msg().version >= 4) {
+    uint32_t fd_count = esapi_->ExecFDCount(&msg.es_msg().event.exec);
+    for (uint32_t i = 0; i < fd_count; i++) {
+      const es_fd_t *fd = esapi_->ExecFD(&msg.es_msg().event.exec, i);
+      pb::FileDescriptor *pb_fd = pb_exec->add_fds();
+      pb_fd->set_fd(fd->fd);
+      pb_fd->set_type(fd->fdtype);
+      std::string_view type_name = GetFileDescriptorTypeName(fd->fdtype);
+      pb_fd->set_type_name(type_name.data(), type_name.length());
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/santad/Logs/EndpointSecurity/Serializers/Protobuf.mm]: Done

File: Source/santad/Logs/EndpointSecurity/Serializers/Protobuf.mm [hunk 6]
```diff
@@ -0,0 +1,609 @@
...
+pb::Execution::Mode GetModeEnum(SNTClientMode mode) {
+  switch (mode) {
+    case SNTClientModeMonitor: return pb::Execution::MODE_MONITOR;
+    case SNTClientModeLockdown: return pb::Execution::MODE_LOCKDOWN;
+    case SNTClientModeUnknown: return pb::Execution::MODE_UNKNOWN;
+    default: return pb::Execution::MODE_UNKNOWN;
+  }
+}
+
+std::string_view GetFileDescriptorTypeName(uint32_t fdtype) {
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/santad/Logs/EndpointSecurity/Serializers/Protobuf.mm]: Done

File: Source/common/santa.proto [hunk 7]
```diff
@@ -11,135 +11,271 @@ import "google/protobuf/timestamp.proto";
...
+  string path = 1;
+  bool truncated = 2;
+  Stat stat = 3;
+  Hash hash = 4;
+}
+
+message FileDescriptor {
+  int32 fd = 1;
+  uint32 type = 2;
+  string type_name = 3;
```
[the remark this answers is not in the document]
Comment (mlw) [reply Source/common/santa.proto]: Done

**Review by pmarkowsky — APPROVED** [review]
LGTM.

**Pull request merged**